libwacom/libwacom-database.c

```diff
@@ -295,32 +412,71 @@ libwacom_parse_stylus_keyfile(WacomDeviceDatabase *db, const char *path)
 
 	groups = g_key_file_get_groups (keyfile, NULL);
 	for (i = 0; groups[i]; i++) {
-		WacomStylus *stylus;
+		WacomStylus *stylus = NULL, *aliased = NULL;
 		WacomStylusId id;
 		GError *error = NULL;
 		char *eraser_type, *type;
 		char **string_list;
+		int aliased_int;
 
 		if (!parse_stylus_id(groups[i], &id)) {
 			g_warning ("Failed to parse stylus ID '%s', ignoring entry", groups[i]);
 			continue;
 		}
+
+		char *aliasstr = g_key_file_get_string(keyfile, groups[i], "AliasOf", NULL);
+
+		if (handle_aliases == IGNORE_ALIASES && aliasstr) {
+			g_clear_pointer(&aliasstr, g_free);
+			continue;
+		}
+
+		if (handle_aliases == ONLY_ALIASES && !aliasstr) {
+			g_clear_pointer(&aliasstr, g_free);
```

Comment:
not really necessary since we explicitly check for it to be NULL but it doesn't hurt and makes it more symmetric I guess.